### ISSUE TITLE
incorrect `indent` behavior for anonymous function

### ISSUE THREAD

**Issue by wli** [eslint/eslint#4174]
Code:
```
             showUpdateContactModal(
               contactType, contactId, function() {
                 selectContact(contactType, contactId); // 339
                 refreshEmergencyForm(); // 340
               }); // 341
```

What eslint wants:
```
             showUpdateContactModal(
               contactType, contactId, function() {
               selectContact(contactType, contactId); // 1339
               refreshEmergencyForm(); // 1340
             }); // 1341
```

Errors:
```
line 1339, col 17, Error - Expected indentation of 14 space characters but found 16. (indent)
line 1340, col 17, Error - Expected indentation of 14 space characters but found 16. (indent)
line 1341, col 15, Error - Expected indentation of 12 space characters but found 14. (indent)
```

Version: eslint 1.7.1

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by wli** [comment]
Possibly related to the fix made in https://github.com/eslint/eslint/issues/4164

**Comment by gyandeeps** [comment]
I am not sure what the right thing here is but if we try to fix the issue above then we have to start marking below example as an error.
```js
require('http').request({hostname: 'localhost',
                         port: 80}, function(res) {
  res.end();
});
```

this is already in our unit test as a valid case: https://github.com/eslint/eslint/blob/master/tests/lib/rules/indent.js#L101-L104

**Comment by gyandeeps** [comment]
Bringing in another use case to add to the original issue: #4184 (since its related)
```js
            foo(
              () => {
                bar();
              }, () => {
                baz();
              }
            );
```

**Comment by silverwind** [comment]
@wli: Your expected output looks not quite right to me, I see no compelling reason to wrap and indent the arguments there. How about something like this:

````js
showUpdateContactModal(contactType, contactId, function() {
    selectContact(contactType, contactId); // 339
    refreshEmergencyForm(); // 340
}); // 341
````

**Comment by wli** [comment]
@silverwind that's a good workaround for this particular case, but it looks like there are competing styles if you have a lot of arguments and you need to line wrap. Basically, this ticket and https://github.com/eslint/eslint/issues/4206 are at odds as to where to put the inner function indent level.

Option A - line length is clearly too long. this should always throw an error.
```
happens.to.be.some.very.very.long.widgetSerializer(one, two, three, four, five six, seven, eight, nine, ten, eleven, function() {
  // function code is here
});
```

Option B - to fit within line length, we indent the arguments at the position of the original argument. It's starting to look rather odd.
```
happens.to.be.some.very.very.long.widgetSerializer(one, two, three,
                                                   four, five six, 
                                                   seven, eight,
                                                   nine, ten,
                                                   eleven, function() {
  // function code here?
                                                     // or here?
});
```

Option C - arguments are no longer at the same indent level, which makes it more difficult to parse
```
happens.to.be.some.very.very.long.widgetSerializer(one, two, three, four,
  five six, seven, eight, nine, ten, eleven, function() {
    // function code here?
  // or here?
  });
```

Option D - you can see all the arguments in one line
```
happens.to.be.some.very.very.long.widgetSerializer(
  one, two, three, four, five six, seven, eight, nine, ten, eleven, function() {
    // function code here?
  // or here?
});
```

**Comment by Pyrolistical** [comment]
@silverwind it might not be valid here, but when you have more levels of nesting, it is valid

**Comment by Pyrolistical** [comment]
I believe the problem is the follow *SHOULD* be considered an error:
```
require('http').request({hostname: 'localhost',
                         port: 80}, function(res) {
  res.end();
});
```

Aligning arbitrary to the open `(` instead of a fixed number of indents is causing the problem.  There is no sane for eslint to figure that out, so we should do the simple thing and require a fixed indent.
